### PR TITLE
add github auth

### PR DESCRIPTION
**Pull request by MichaelPetrinolis**
add github auth to the list of supported external providers

### CODE REVIEW

**Comment by agriffard** [comment]
Thank you Michael.

**Comment by MichaelPetrinolis** [comment]
Check it out when you get some time...

**Comment by agriffard** [comment]
OK, I managed to: 
Create an OAuth application on github
Enable the github authentication module
Set the client Id and Secret

Sign in with Github and link to an existing account

Enable registration
Register as another user with github

I am wondering a few things:
The association is only based on the username? The email is not retrieved from the github account when you register?
When you edit it in the admin, a user with external logins does not have any additional informations displayed?

**Comment by MichaelPetrinolis** [comment]
1. If a user tries to login with external provider, and registration is enabled then 
- If the external provider provides an email claim, we search for an existing account with that email
- If not found, a new OC Account is created based on the username,email and password provided by the user
- If found, we request the password of the OC Account in order to link the external login
There is a PR that takes into consideration the email must confirmed parameter, if we request password in order to create a local Account from external login, and also uses the recaptcha

2. There is a front page (OrchardCore.Users/Account/ExternalLogins) where you can link/unlink OC account with external providers. Of course, is not complete, we must decide how this is integrated with profile/account info in admin/front end

**Comment by agriffard** [comment]
Fixes #3483

File: src/OrchardCore.Modules/OrchardCore.Github/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Github (OrchardCore.Github)
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Github/README.md]: Missing the entry in the main documentation index?

Comment (sebastienros) [reply src/OrchardCore.Modules/OrchardCore.Github/README.md]: ping

Comment (MichaelPetrinolis) [reply src/OrchardCore.Modules/OrchardCore.Github/README.md]: Didn't know how to do it. Found it in mkdocs.yml

File: src/OrchardCore.Modules/OrchardCore.Github/Recipes/TwitterSigninSettingsStep.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+using OrchardCore.Recipes.Services;
+
+namespace OrchardCore.Github.Recipes
+{
+    /// <summary>
+    /// This recipe step sets Microsoft Account settings.
+    /// </summary>
+    public class GithubSigninSettingsStep : IRecipeStepHandler
+    {
+        private readonly IGithubAuthenticationService _twitterLoginService;
```
Comment (Icidis) [inline src/OrchardCore.Modules/OrchardCore.Github/Recipes/TwitterSigninSettingsStep.cs]: Variable name does not match Interface and class ```<summary>``` is also incorrect.

Comment (Icidis) [reply src/OrchardCore.Modules/OrchardCore.Github/Recipes/TwitterSigninSettingsStep.cs]: Also the class should be changed to GithubSigninSettingsStep.cs

Comment (MichaelPetrinolis) [reply src/OrchardCore.Modules/OrchardCore.Github/Recipes/TwitterSigninSettingsStep.cs]: thanks @Icidis , the PR is already merged, so you can open a new one if you want to contribute, or tell me to do it.

**Pull request merged**